qutebrowser/misc/editor.py

```diff
@@ -135,8 +135,6 @@ def edit(self, text, caret_position=None):
             message.error("Failed to create initial file: {}".format(e))
             return
 
-        self._remove_file = True
```

Comment:
I'd suggest to only change this based on the setting, and leave the others untouched - right now, for example, the file isn't removed when the editor failed to start (if I remember correctly), while your new logic would always remove the file if the setting is set.